File: countDown/date.c
#include <stdio.h>
#include <time.h>
#include <windows.h>

int main() {
    int day, hour, minute;
    time_t current_time, future_time;
    struct tm *future_tm;
    char input_str[10];

    // Prompt the user to input a future time point
    printf("Please enter a future time point (format: day-hour-minute, e.g. 05-17-30): ");
    scanf("%d-%d-%d", &day, &hour, &minute);

    // Use the current date and time to create a time_t object
    current_time = time(NULL);

    // Use the current date and time and the user input to calculate the future time point as a time_t object
    future_tm = localtime(&current_time);
    future_tm->tm_mday = day;
    future_tm->tm_hour = hour;
    future_tm->tm_min = minute;
    future_time = mktime(future_tm);

    while (1) {
        // Get the current date and time
        current_time = time(NULL);

        // Calculate the time difference between the current time and the future time point
        double time_left = difftime(future_time, current_time);

        // Convert the time difference to hours, minutes, and seconds
        int hours = (int) time_left / 3600;
        int minutes = (int) (time_left - hours * 3600) / 60;
        int seconds = (int) time_left % 60;

        // Print the time difference in the format of hours:minutes:seconds
        printf("Time left until the future time point: %02d:%02d:%02d\n", hours, minutes, seconds);

        // Wait for 5 seconds before updating the time difference again
        Sleep(5000);
    }

    return 0;
}

File: countDown/date.go
package main

import (
    "fmt"
    "strings"
    "strconv"
    "time"
)

func main() {
    // 提示用户输入未来时间点
    fmt.Println("请输入未来时间点（格式：日子-小时-分钟，例如：05-17-30）：")
    var inputStr string
    fmt.Scanln(&inputStr)

    // 解析用户输入的字符串为日、小时和分钟
    input := strings.Split(inputStr, "-")
    day, _ := strconv.Atoi(input[0])
    hour, _ := strconv.Atoi(input[1])
    minute, _ := strconv.Atoi(input[2])

    // 使用当前日期和时间，以及用户输入的日、小时和分钟创建时间对象
    currentTime := time.Now()
    futureTime := time.Date(currentTime.Year(), currentTime.Month(), day, hour, minute, 0, 0, currentTime.Location())

    for {
        // 获取当前日期和时间
        currentTime = time.Now()

        // 计算距离未来时间点的时间差
        timeLeft := futureTime.Sub(currentTime)

        // 格式化时间差为时:分:秒
        hours := int(timeLeft.Hours())
        minutes := int(timeLeft.Minutes()) % 60
        seconds := int(timeLeft.Seconds()) % 60
        timeLeftStr := fmt.Sprintf("%02d:%02d:%02d", hours, minutes, seconds)

        // 打印时间差
        fmt.Println("Time left until the future time point:", timeLeftStr)

        // 每隔5秒钟更新一次时间差
        time.Sleep(5 * time.Second)
    }
}
